framework/src/service/ServiceReferenceBasePrivate.cpp

```diff
@@ -100,11 +102,13 @@ InterfaceMapConstPtr ServiceReferenceBasePrivate::GetServiceFromFactory(
     s = smap;
   } catch (...) {
     s.reset();
+    std::string message = "ServiceFactory threw an unknown exception.";
     registration->bundle->coreCtx->listeners.SendFrameworkEvent(
       FrameworkEvent(FrameworkEvent::Type::FRAMEWORK_ERROR,
                      MakeBundle(bundle->shared_from_this()),
-                     std::string("ServiceFactory threw an unknown exception."),
-                     std::current_exception()));
+                     message,
+                     std::make_exception_ptr(ServiceException(message,
```

Comment:
Since an exception was thrown, we want to preserve the exception message and use that string as the ```message``` in the ```ServiceException``` object. Otherwise we are losing information which may be useful to users.